test/quantization/test_quantize_fx.py

```diff
@@ -961,6 +964,52 @@ def forward(self, x):
         # quantize, should run with no errors
         quantized = convert_fx(prepared_copy)
 
+    def test_fx_traceable_marker(self):
+        """Tests that the QuantizationFXTraceable marker works correctly."""
+        class Child(nn.Module, QuantizationFXTraceable):
+            def __init__(self):
+                super().__init__()
+                self.conv = torch.nn.Conv2d(1, 1, 1)
+
+            def forward(self, x):
+                x = self.conv(x)
+                return x
+
+        class M(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.child = Child()
+
+            def forward(self, x):
+                # non traceable
+                if x.dim() == 1:
+                    raise NotImplementedError
+                # traceable
+                x = self.child(x)
+                return x
+        m = M()
+        m.eval()
+
+        # ensure that the parent is not symbolically traceable
+        with self.assertRaises(torch.fx.proxy.TraceError):
+            gm = torch.fx.symbolic_trace(m)
+
+        # prepare
+        qconfig_dict = {'': torch.quantization.default_qconfig}
+        prepare_custom_config_dict = {'use_fx_traceable_marker': True}
```

Comment:
looks like we have two modes: 
'use_fx_trraceable_marker':False --> default traceable, user should blacklist non traceable parts
'use_fx_traceable_marker': True --> default non traceable, user should whitelist traceable parts.

my question is, should we have both modes? if so, should we also have a `QuantizationFXNonTraceable` for consistency?

Reply:
> my question is, should we have both modes?

I would go for two modes.  This PR only implements one, but there is a TODO in the code to also add a marker class for leaves.

The intent of the `use_fx_traceable_marker` flag is add a way to turn off this functionality.  One of the things to discuss would be if we actually need this flag, or if we can assume that if a marker is present it must be accounted for.  One thing to consider is a reusable building block which may be used in different contexts, which is why my first proposal has this flag, and the user has the ability to ignore the markers.